### ISSUE TITLE
Unsupported OpenGL Extensions macOS

### ISSUE THREAD

**Issue by YanvegHD** [yuzu-emu/yuzu#538]
Hello,

I'm trying to run yuzu emu on my MacBook Pro (15 inch early 2013, with Nvidia GT650M), and I can't get it working, I'm trying to run a hello world NRO homebrew, here's the error I get : "Your GPU may not support one or more required OpenGL extensions. Please ensure you have the latest graphics driver. See the log for more details."

And then in the console log : 

[   0.000000] Frontend <Info> yuzu/main.cpp:BootGame:448: yuzu starting...
[   0.066674] Frontend <Critical> yuzu/main.cpp:SupportsRequiredGLExtensions:351: Unsupported GL extension: ARB_program_interface_query
[   0.066698] Frontend <Critical> yuzu/main.cpp:SupportsRequiredGLExtensions:351: Unsupported GL extension: ARB_shader_storage_buffer_object
[   0.066706] Frontend <Critical> yuzu/main.cpp:SupportsRequiredGLExtensions:351: Unsupported GL extension: ARB_vertex_attrib_binding

Is it because my hardware is too old or I have to wait for an update ? Thanks ! Good luck with dev :)

**Comment by legoj15** [comment]
A standard GT650M claims to support up to GL 4.5, but since you're using a MacBook, there's a really high chance that it's Apple's fault for having cruddy drivers for OpenGL. There may be a possibility in the coming years if yuzu decides to add/switch over to Vulkan as a backend, which can be "used" on a Mac (technically, through MoltenVK).
TL;DR, *maybe* in a few years it could work, but don't bet on it, since your mac is old and Apple hates everyone who doesn't use their Metal backend instead of OpenGL now.

**Comment by MoochMcGee** [comment]
This is why you shouldn't use a Mac.

**Comment by Plnda** [comment]
Are you on Mojave? got the same issue where before it just ran fine

**Comment by mailwl** [comment]
same issue - nintendo switch works on opengl 4.6, macmini on 4.1. that sukcs

**Comment by Haaroon** [comment]
issue stil pending on 10.13.3

**Comment by MoochMcGee** [comment]
@Haaroon Do you not read? People have clearly stated that this problem is Apple's fault, not yuzu's.

**Comment by legoj15** [comment]
I'm 99.97% sure that the reason old builds """"worked"""" is because there were no OGL checks inplace. The last month and a half of builds now have checks to make sure yuzu can actually try to output 3D, instead of just blindly opening and then crashing when the application requests ARB_ extensions from the OpenGL driver that don't exist.

**Comment by pcbeard** [comment]
Has anybody considered moving the Mac port to Metal? How deep are the OpenGL dependencies? This site has some guidance for porting from OpenGL to Metal:

https://www.haroldserrano.com/metal-api-projects/

**Comment by mailwl** [comment]
@pcbeard, yuzu is open source project, if you need some feature - just implement it.
not sure any developers care about macos, sorry

**Comment by majuss** [comment]
Try the Nvidia webdrivers, maybe it helps :)
https://www.tonymacx86.com/nvidia-drivers/

**Comment by majuss** [comment]
Could someone try the nvidia webdrivers? I believe they come with a different openGL version.

**Comment by iSn0we** [comment]
tried. failed.

**Comment by iSn0we** [comment]
Will this be fixed anywhen?

**Comment by FernandoS27** [comment]
Why is this marked as a bug? This is just a problem that’s not yuzu’s fault, nor is fixable from their end.

For people asking, this can only be fixed by Apple. Yuzu requires a series of extensions and features that Apple must provide on their drivers. 

If you want a solution to keep using your hardware (Your .Mac) Install a dual boot OS with Ubuntu and run yuzu on Linux

**Comment by Hexagon12** [comment]
Requesting to close.

**Comment by MerryMage** [comment]
Closed as Won’t Fix.

Apple have deprecated OpenGL support, so it is unlikely yuzu will support macOS for the foreseeable future.

**Comment by majuss** [comment]
Any plans on support Vulkan? This would get macOS back on board.

**Comment by shianiawhite** [comment]
> Any plans on support Vulkan? This would get macOS back on board.

Yes, here is [the pull request](https://github.com/yuzu-emu/yuzu/pull/1114). And yes, this should make macOS viable again through MoltenVK.
